check/src/kindcheck.rs

```diff
@@ -143,7 +120,7 @@ impl<'a> KindCheck<'a> {
     fn builtin_kind(&self, typ: BuiltinType) -> RcKind {
         match typ {
             BuiltinType::String | BuiltinType::Byte | BuiltinType::Char | BuiltinType::Int |
-            BuiltinType::Float | BuiltinType::Unit => self.type_kind(),
+            BuiltinType::Float | BuiltinType::Unit | BuiltinType::Type => self.type_kind(),
```

Comment:
See https://github.com/Marwes/gluon/pull/131#discussion_r76334248. I think at least as inference works now (types, kinds etc are all separate, only the representation is unified), this should be fine. If that is true maybe it is fine to omit universe variables for now.